sessions_core/01_introduction.qmd

```diff
@@ -0,0 +1,751 @@
+
+#### Simple Vectors
+Simple vectors can contain one or more values of a single data type, they thus have two key properties: **length** and **type**. For the purpose of this class, we will use the terms “simple vector” and “vector” interchangeably (as is typical in the R community). 
+
+You’ve technically already created your very first simple vectors when you built `cases` and `region`. These were simply vectors with a **length of one**. To create a vector with more than one value, we will use the function [`c()`]{.hoverinfo toggle='tooltip' title="Mnemonic: `c` is the first letter of “concatenate”"}:
+
+```{r}
+#| eval: true
+cases <- c(2, 5, 8, 0, 4)
+```
+
+::: {.write}
+Update `cases` with the above values and update `region` to create a string vector containing the values: Mandoul, Moyen-Chari, Logone Oriental, Tibesti, Logone Occidental.
+:::
+
+```{r}
+#| include = solutions
+# SOLUTION
+region <- c("Mandoul", "Moyen-Chari", "Logone Oriental",
+                      "Tibesti", "Logone Occidental")
+```
+
+```{r}
+#| echo: false
+#| eval: true
+# INTERNALS
+region <- c("Mandoul", "Moyen-Chari", "Logone Oriental",
+                      "Tibesti", "Logone Occidental")
+```
+
+
+We can now use functions on the objects we have created:
+
+```{r}
+#| eval: true
+mean(cases)      # calculate the mean value of the cases vector
+toupper(region)  # convert all the values in region to lower case
+```
+
+
+::: {.write}
+Use the function `sum()` to calculate the sum of `cases`
+:::
+
+```{r}
+#| include = solutions
+# SOLUTION
+sum(cases)
+```
+
+::: {.write}
+Pass `region` to the function `tolower()`
+:::
+
+```{r}
+#| include = solutions
+# SOLUTION
+tolower(region)
+```
+
+### Accessing the Values of a Vector
+
+It is possible to access the value of a vector using square brackets containing the index (position) of the desired value, ie: `[3]` or `[189]`.
+
+```{r}
+#| eval: true
+cases[2]   # 2nd value of cases
+cases[10]  # 10th value of cases
+```
+
+Ooops it does not exist! We will come back to what this `NA` means in @sec-missing-values.
+
+We can also access a *range* of values, just as we might do in Excel. To create a range we use the `:` operator to separate the desired minimum and maximum index:
+
+```{r}
+#| eval: true
+cases[2:4]  # 2nd to 4th values of cases
+```
+
+::: {.write}
+Get the 3rd value of `region`. Write code to access the values "Mandoul" and "Moyen-Chari" in the vector `region`.
+:::
+
+```{r}
+#| include = solutions
+# SOLUTION
+region[3]
+region[1:2]
+```
+
+
+### Dataframes
+
+*Dataframes* are a special type of list that is composed of multiple simple vectors of the same length. Dataframes take the shape of a table where each of these vectors is a named column, similar to a table in Excel. As epidemiologists, this type of data structure is perhaps the most useful and you will likely use them on a daily basis.
```

Comment:
I personally find the first sentence of this definition too abstract for an applied course. I'd lead with them being a 2d / tabular structure with rows and named columns, where the columns can be of different types. Could also give an example of this that they would have seen (e.g. a linelist).